internal/app/accrualdaemon/accrual_daemon.go

```diff
@@ -0,0 +1,154 @@
+package accrualdaemon
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go.uber.org/zap"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+	"yandex_gophermart/pkg/entities"
+	gophermart_errors "yandex_gophermart/pkg/errors"
+)
+
+//
+//const (
+//	orderStatusRegistered = "REGISTERED"
+//	orderStatusInvalid    = "INVALID"
+//	orderStatusProcessing = "PROCESSING"
+//	orderStatusProcessed  = "PROCESSED"
+//)
+
+type UnfinishedOrdersStorageInt interface {
+	UpdateOrder(orderData entities.OrderData, ctx context.Context) error
+	GetUnfinishedOrdersList(ctx context.Context) ([]entities.OrderData, error)
+	AddToBalance(userID int, amount float64, ctx context.Context) error
+}
+
+type respData struct {
+	Order   string  `json:"order"`
+	Status  string  `json:"status"`
+	Accrual float64 `json:"accrual"`
+}
+
+func AccrualCheckDaemon(ctx context.Context, logger *zap.SugaredLogger, storage UnfinishedOrdersStorageInt, accrualSystemAddress string) {
+	logger.Infof("Accrual daemon started")
+
+	orders := make([]entities.OrderData, 0)
+	i := 0
+
+loop:
+	for {
+		//get new unfinished orders
+		if i >= len(orders) {
+			var err error
+			orders, err = storage.GetUnfinishedOrdersList(ctx)
+			if err != nil {
+				logger.Errorf("cant get unfinished orders from db, err: %v", err.Error())
+				return
+			}
+			i = 0
+		}
+
+		select {
+		case <-ctx.Done():
+			break loop
```

Comment:
```suggestion
			return
```
Мы все равно выходим из цикла в конец функции. Нет смысла городить конструкцию с break.

Reply:
Fix